### ISSUE TITLE
youtube.com

### ISSUE THREAD

**Issue by Morku** [AdguardTeam/AdguardFilters#170429]
### Prerequisites

- [X] This site DOES NOT contain sexually explicit material, otherwise use [NSFW-specific form](./new?template=bug_report_NSFW.yml);
- [X] Filters were updated before reproducing an issue;
- [X] AdGuard product version is up-to-date;
- [X] Browser version is up-to-date;
- [X] If the site or app is broken, disabling AdGuard protection resolves an issue.

### What product do you use?

AdGuard for Windows

### AdGuard version

7.16.0 (4542)

### What type of problem have you encountered?

Annoyances — pop-ups, cookie warnings, etc.

### Which browser(s) do you use?

Firefox

### Which device do you use?

Desktop

### Where is the problem encountered?

https://www.youtube.com/

### Ad Blocking

AdGuard Base filter, AdGuard Mobile Ads filter

### Privacy

AdGuard URL Tracking filter, AdGuard Tracking Protection filter

### Social Widgets

AdGuard Social Media filter

### Annoyances

AdGuard Cookie Notices filter, AdGuard Popups filter, AdGuard Mobile App Banners filter, AdGuard Other Annoyances filter, AdGuard Widgets filter

### Language-specific

AdGuard German filter

### What Stealth Mode options do you have enabled?

Hide your search queries, Send Do-Not-Track signals, Remove X-Client-Data header from HTTP requests

### Add your comment and screenshots

It looks like ticket was misunderstood: https://github.com/AdguardTeam/AdguardFilters/issues/170326

This report is about an unblocked Cookie Pop-up that is normaly handled by this filterlist: https://github.com/AdguardTeam/AdguardFilters/tree/master/AnnoyancesFilter/Cookies/sections

I did not mention any history whatever profil thing. It's about the Cookie Pop-up

Currently it is not filtered. My previous ticket even has two Thumb up as other users have the same issue.

<details><summary>Screenshot 1:</summary>

![youtube](https://github.com/AdguardTeam/AdguardFilters/assets/6276915/0323523d-f099-4de1-a116-aceb4d99e00f)

<details><summary>Screenshot N:</summary>

</details>

### Privacy

- [X] I agree to follow this condition

**Comment by Alex-302** [comment]
If rejected, history can't be used

![image](https://github.com/AdguardTeam/AdguardFilters/assets/8361299/eeacd453-2b7d-4dd5-a88f-21f2adea0a55)

**Comment by Morku** [comment]
@Alex-302 

Thats wrong. If **you** want YouTube History without Profile, you can click Update setting and click On afterwards.

![296175702-eeacd453-2b7d-4dd5-a88f-21f2adea0a55](https://github.com/AdguardTeam/AdguardFilters/assets/6276915/0a6ab40f-7277-4461-bd00-272c9a4630ce)

![Screenshot 2024-01-12 094042](https://github.com/AdguardTeam/AdguardFilters/assets/6276915/944b34be-709b-4916-9944-73d3ddac62d8)


Since when AdGuard has become a not privacy focused product anymore?

For years those Cookie Pop-up were autoclicked, giving the user the choice **afterwards** to enable e.g. embedded Twitter, Youtube, Instagram.
This Pop-up is exactly the same and that is one purpose why people install AdGuard.

@AdamWr 
Can you please check? Discussing with Alex is tiring.

**Comment by AdamWr** [comment]
This rule:
```adblock
youtube.com#%#//scriptlet('trusted-click-element', 'ytd-consent-bump-v2-lightbox .eom-buttons > .ytd-consent-bump-v2-lightbox > .ytd-consent-bump-v2-lightbox button.yt-spec-button-shape-next')
```
should automatically reject cookie notice.

This one:
```adblock
youtube.com#%#//scriptlet('trusted-click-element', 'ytd-consent-bump-v2-lightbox .eom-buttons > .ytd-consent-bump-v2-lightbox > .ytd-consent-bump-v2-lightbox:nth-of-type(2) button.yt-spec-button-shape-next')
```
should automatically accept it.

Rejecting cookie is probably better for privacy, but I guess that it might be a bit annoying for some users that they have to change settings to see content on the main page.

**Comment by Alex-302** [comment]
@AdamWr With rejecting, history can't be used.
Maybe just apply? In private tab cookies are not so important, I think.

**Comment by Morku** [comment]
@AdamWr 
I can confirm, both rules working.

IMO:
User who decide to install AdGuard gets what they want. And if subscribing Cookie filter and privacy filter, **my** expectation is that Pop-up gets rejected.
And people are always complaining about youtube, by false using AdGuard or expect that App gets fiiltered. I wouldn't mind.

I still believe in intelligence in people to read the notification why history is turned off and click **Update settings** to turn it On. If they save Cookies -> it stays On.
Most people Log in into profile anyway and will adjust the profile to save history in profile.

But there are also a bunch of users who always browse in private mode for a reason and want share data on minimum (reason for isntalling AG...)

tldr, it's up to you.

@Alex-302 
You can enable history if you like to, see post above.

**Comment by AdamWr** [comment]
>With rejecting, history can't be used.

If I'm not wrong, it can be used only if user is logged and even if cookies are rejected it depends on user's settings (not related to cookie notice).

**Comment by Morku** [comment]
Idea:

Is it possible to check depency combine filter subscriptions?

Users who subscribed Tracking filter -> Pop-up gets rejected
User who don't subscribe Tracking filter -> Pop-up gets accepted

**Comment by Alex-302** [comment]
@Morku 

> You can enable history if you like to, see post above.

![image](https://github.com/AdguardTeam/AdguardFilters/assets/8361299/711f5d48-f3a2-43c1-abc9-222562eaa081)

**Comment by Morku** [comment]
![grafik](https://github.com/AdguardTeam/AdguardFilters/assets/6276915/74418bbb-da64-4e07-b700-f2e572b80e44)

**Comment by Alex-302** [comment]
@Morku Confirm = Accept

**Comment by Alex-302** [comment]
> If I'm not wrong, it can be used only if user is logged and even if cookies are rejected it depends on user's settings (not related to cookie notice).

Hm. You are right.

**Comment by AdamWr** [comment]
So, maybe we can add a rule to reject cookies and if there will be complains about that, then we could change it to accept cookies.

**Comment by Alex-302** [comment]
@AdamWr ok

**Comment by AdamWr** [comment]
Done.

**Comment by Morku** [comment]
@AdamWr 
I see you have added a rule for the mobile version. But there is another(?) Pop-up that is not autoclicked with that rule on iPhone or Desktop mobile view (iPhone).
Can you please check?

https://m.youtube.com/watch?v=yasSmePUpJc

<details><summary>Screenshot-1:</summary>

![yt](https://github.com/AdguardTeam/AdguardFilters/assets/6276915/4bc7105f-3eda-43db-bf4b-153ceaf67555)


</details><br/>

**Comment by AdamWr** [comment]
I guess that the problem is that it's clicked too fast and some scripts are not loaded or for some another reasons it's not closed.

Could you please test this rule:
```adblock
m.youtube.com#%#//scriptlet('trusted-click-element', 'ytm-consent-bump-v2-renderer .dialog-scrollable-content > .one-col-dialog-buttons > ytm-button-renderer.eom-reject > button.yt-spec-button-shape-next', '', '1000')
```
If it doesn't work please test this one:
```adblock
m.youtube.com#%#//scriptlet('trusted-click-element', 'ytm-consent-bump-v2-renderer .dialog-scrollable-content > .one-col-dialog-buttons > ytm-button-renderer.eom-reject > button.yt-spec-button-shape-next', '', '1500')
```
and if still then this one:
```adblock
m.youtube.com#%#//scriptlet('trusted-click-element', 'ytm-consent-bump-v2-renderer .dialog-scrollable-content > .one-col-dialog-buttons > ytm-button-renderer.eom-reject > button.yt-spec-button-shape-next', '', '2000')
```

**Comment by Morku** [comment]
@AdamWr 

`1000` almost never worked.
`1500` was mostly fine on my iPhone XS, but still like 50% failure on my PC (which isn't obvious important)
`2000` was fine for both devices and looks even quick on my iPhone. It failed only once on my PC. But my PC is also currently busy with encodings.

**Comment by AdamWr** [comment]
Okay I have added rule with 2 seconds delay.
Thank you for checking it out.

**Comment by Morku** [comment]
@AdamWr 

Google changed the Cookie Pop-up for mobile when browsing a video directly. So it does not autoclick anymore (also verified on iPhone). I highlighted the change in Screenshot. Can you please check?

<details><summary>Screenshot-1:</summary>

![yt](https://github.com/user-attachments/assets/202f0371-8fa9-418b-97f2-eadb40c0f5bb)


</details><br/>

**Comment by AdamWr** [comment]
Okay, I have updated the rule.
